pkg/resource/deploy/step_generator.go

```diff
@@ -72,8 +72,34 @@ func (sg *stepGenerator) isTargetedUpdate() bool {
 	return sg.updateTargetsOpt != nil || sg.replaceTargetsOpt != nil
 }
 
-func (sg *stepGenerator) isTargetedForUpdate(urn resource.URN) bool {
-	return sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn]
+// isTargetedForUpdate returns if urn is target for update. The function accommodates
+// `--target-dependents`. `targetDependentsForUpdate` should probably be called if this function
+// returns true.
+func (sg *stepGenerator) isTargetedForUpdate(
+	urn, parent resource.URN, provider string, dependencies []resource.URN) bool {
```

Comment:
Nit: maybe use a single `*resource.State` parameter and pull properties off of that:

```suggestion
func (sg *stepGenerator) isTargetedForUpdate(r *resource.State) bool {
```